test/extensions/common/wasm/test_data/test_restriction_cpp.cc

```diff
@@ -0,0 +1,25 @@
+#include <cstdio>
+#include <cstdlib>
+#include <limits>
+#include <string>
+
+#ifndef NULL_PLUGIN
+#include "proxy_wasm_intrinsics.h"
+#else
+#include "include/proxy-wasm/null_plugin.h"
+#endif
+
+START_WASM_PLUGIN(CommonWasmRestrictionTestCpp)
+
+WASM_EXPORT(void, proxy_abi_version_0_2_1, (void)) {}
+
+WASM_EXPORT(uint32_t, proxy_on_vm_start, (uint32_t context_id, uint32_t configuration_size)) {
+  (void)(context_id);
+  (void)(configuration_size);
+  std::string level_message = "after on_vm_start, before proxy_log";
```

Comment:
Nit: `s/level_message/log_message/`

Also, this message is written as part of `proxy_log`, not before `proxy_log`.

Reply:
Fixed.